From 628620a1c0f8642b0fc2cf442da92566bc7d468b Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Thu, 15 Oct 2020 17:39:42 +0200
Subject: [PATCH] PONRTSYS-8120: Simplify the code

sch is set to root->children[i] some lines before and the NULL check is done on sch.
---
 drivers/net/datapath/pon_qos/pon_qos_tc_qos.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/datapath/pon_qos/pon_qos_tc_qos.c b/drivers/net/datapath/pon_qos/pon_qos_tc_qos.c
index 9a863f7a6266..2fed69c54cea 100644
--- a/drivers/net/datapath/pon_qos/pon_qos_tc_qos.c
+++ b/drivers/net/datapath/pon_qos/pon_qos_tc_qos.c
@@ -693,7 +693,7 @@ int pon_qos_qdisc_tree_del(struct pon_qos_port *p, struct pon_qos_qdisc *root)
 		if (sch && sch->use_cnt) {
 			netdev_dbg(sch->dev, "%s: delete child %u of %u\n",
 				   __func__,
-				   root->children[i]->sch_id,
+				   sch->sch_id,
 				   root->sch_id);
 			ret = pon_qos_qdisc_tree_del(p, sch);
 			if (ret < 0)
